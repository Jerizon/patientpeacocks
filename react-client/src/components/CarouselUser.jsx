import React from 'react';
import Button from 'react-bootstrap/lib/Button';
import ButtonToolbar from 'react-bootstrap/lib/ButtonToolbar'
import Carousel from 'react-bootstrap/lib/Carousel'

class CarouselPageUser extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      direction: null
    }
    this.handleSelect.bind(this);
  }

  handleSelect(selectedIndex, e) {
    var context = this;
    context.setState({
      index: selectedIndex,
      direction: e.direction
    });
  }

  render() {
    return (
      <Carousel className = 'carousel' activeIndex={this.state.index} direction={this.state.direction} onSelect={this.handleSelect.bind(this)}>
        <Carousel.Item>
          <img width={2000} height={2000} alt="2000x2000" src="https://voltcommerce.com/news/wp-content/uploads/2014/04/Coachella_Night_Pano_2-FINAL_Flat.jpg" />
          <Carousel.Caption>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <img width={2000} height={2000} alt="2000x2000" src="http://www.visitdenmark.com/sites/default/files/styles/block_ratio/public/vdk_images/Attractions-Activities-interest-accommodation-people-geo/Events/Sporting-events/color-run-credit-colorrun.dk.jpg?itok=udG3wZA3" />
          <Carousel.Caption>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <img width={2000} height={2000} alt="2000x2000" alt="300x1000" src="http://cdn.funcheap.com/wp-content/uploads/2014/07/fireworks21.jpg" />
          <Carousel.Caption>
          </Carousel.Caption>
        </Carousel.Item>
      </Carousel>
    );
  }
}
export default CarouselPageUser;