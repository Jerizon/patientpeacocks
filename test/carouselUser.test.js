import React from 'react';
import ReactDOM from 'react-dom';
import CarouselUser from '../react-client/src/components/CarouselUser.jsx';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: App', () => {
  const items = ['Learn react', 'rest', 'go out'];

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <CarouselUser/>
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});