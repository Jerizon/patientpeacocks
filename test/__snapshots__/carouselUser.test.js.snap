// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div
  className="carousel carousel slide"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      className="item active"
    >
      <img
        alt="2000x2000"
        height={2000}
        src="https://voltcommerce.com/news/wp-content/uploads/2014/04/Coachella_Night_Pano_2-FINAL_Flat.jpg"
        width={2000}
      />
      <div
        className="carousel-caption"
      />
    </div>
    <div
      className="item"
    >
      <img
        alt="2000x2000"
        height={2000}
        src="http://www.launchphotography.com/AT_T_Park_Panorama.jpg"
        width={2000}
      />
      <div
        className="carousel-caption"
      />
    </div>
    <div
      className="item"
    >
      <img
        alt="300x1000"
        height={2000}
        src="http://cdn.funcheap.com/wp-content/uploads/2014/07/fireworks21.jpg"
        width={2000}
      />
      <div
        className="carousel-caption"
      />
    </div>
  </div>
  <a
    className="carousel-control left"
    href="#"
    onClick={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-left"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control right"
    href="#"
    onClick={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-right"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
